subtests/docker_cli/attach/attach.py

```diff
@@ -36,10 +36,34 @@ def wait_interactive_cmd(self):
                      % wait)
         time.sleep(wait)
 
+    def image_exist(self, image_name):
+        di = DockerImages(self)
+        image_list = DockerImages.get_dockerimages_list(di)
+        exist_status = DockerImages.filter_list_full_name(image_list,
+                                                          image_name)
+        return exist_status
+
+    def pull_image(self, image_name):
+        print "pull_image"
+        dkrcmd = AsyncDockerCmd(self, 'pull', [image_name],
+                                self.config['docker_timeout'],
+                                verbose=True)
+        self.loginfo("Executing background command: %s" % dkrcmd)
+        dkrcmd.execute()
+        while not dkrcmd.done:
+            self.loginfo("Pulling...")
+            time.sleep(3)
+        result = dkrcmd.wait()
+        self.failif(result.exit_status != 0,
+                    "Fail to download image %s"
+                    % image_name)
+
     def initialize(self):
         super(attach_base, self).initialize()
         self.sub_stuff['subargs'] = self.config['run_options_csv'].split(',')
         fin = DockerImage.full_name_from_defaults(self.config)
+        if self.image_exist(fin) == []:
+            self.pull_image(fin)
```

Comment:
Oooohhh! I see what you're doing here now WRT e-mail on the subject, it's not just for this one test right?  Oh man, duhhhhh :blush: now I understand!   We **DO** have an API for this!  Adding this stuff as a new "test" (same format) as a [*pretest*](https://github.com/autotest/autotest-docker/blob/master/pretests/pretest_example/pretest_example.py) and/or [*intratest*](https://github.com/autotest/autotest-docker/blob/master/intratests/intratest_example/intratest_example.py]).

A *pretest* will run before any other tests (unless excluded), and an *intratest* will run between every other test.  Then you can also have a single configuration file (or default) that says: ``pull_missing_test_fqin = yes`` and globally control this behaviour for every test that's about to run.

Also, tell me if you think I should [include running all *intratest* after *pretest*](https://github.com/autotest/autotest-docker/compare/autotest:master...cevich:preintra) but before first test.  I'm thinking this would help avoid duplicating same 'pull' test code in both.

Otherwise, if I'm wrong and you just want to fix this one test, it's fine also :smile: